tools/monitoring/mysql/db_mysql.go

```diff
@@ -118,3 +118,45 @@ func PubsubMsgHandler(db *sql.DB, configURL, buildLogURL, jobname string, prNumb
 
 	return tx.Commit()
 }
+
+// CheckAlertCondition checks whether the given error pattern meets
+// the alert condition specified in config
+func CheckAlertCondition(errorPattern string, config *config.SelectedConfig, db *sql.DB) (bool, error) {
+	// the timestamp we want to start collecting logs
+	startTime := time.Now().Add(time.Minute * time.Duration(config.Period))
+
+	_, err := db.Query(
+		"CREATE VIEW Matched AS\nSELECT Jobname, PrNumber FROM ErrorLogs\nWHERE\n    ErrorPattern=? and\n    TimeStamp > ?",
+		errorPattern, startTime)
+
+	if err != nil {
+		return false, err
+	}
+
+	var count int
+	if err = db.QueryRow("SELECT COUNT(*) FROM Matched;").Scan(&count); err != nil {
+		return false, err
+	}
+
+	if count < config.Occurrences {
```

Comment:
In my opinion, I think updating `count` in after every `select` makes the code hard to read. I think you can combined the `select` statements and do `select count(*) as all_errors, count(distinct jobname) as failed_jobs, ...` and stores the count in three different vars and do a check at the end.